src/org/traccar/protocol/SviasProtocolDecoder.java

```diff
@@ -0,0 +1,150 @@
+            .text("[")       // delimiter init
+            .number("(dddd),") // version hardware
+            .number("(dddd),") // version software
+            .number("(d+),") // counter
+            .number("(d+),") // imei
+            .any()            // model or hourmeter
+            .number("(d+),") // date (yyyymmdd)
+            .number("(d+),") // time (hhmmss)
+            .number("(-?d+),") // longitude
+            .number("(-?d+),") // latitude
+            .number("(d+),") // speed
+            .number("(d+),") // course
+            .number("(d+),") // odometer
+            .number("(d+),") // input
+            .number("(d+),") // output / status
+            .number("(d+),") // flag pack input 1
+            .number("(d+),") // flag pack input 2
+            .number("(d+),") // main power voltage
+            .number("(d+),") // percentual power internal battery
+            .number("(d+),") // RSSID
+            .any()
+            .compile();
+
+    private double convertCoordinates(long v) {
+        return Double.valueOf(((float) ((((float) v / 1.0E7F)
+                - ((int) (v / 10000000L))) * 1.6666666666666667D)) + ((int) (v / 10000000L)));
+    }
+
+    @Override
+    protected Object decode(Channel channel, SocketAddress remoteAddress, Object msg)
+            throws Exception {
+
+        String sentence = (String) msg;
+
+        if (!sentence.contains(":")) {
+
+            Parser parser = new Parser(PATTERN, (String) sentence);
+            if (!parser.matches()) {
+                return null;
+            }
+
+            Position position = new Position(getProtocolName());
+
+            String versionHard = parser.next();
+            String versionSoft = parser.next();
+            String counterInternal = parser.next();
+
+            DeviceSession deviceSession = getDeviceSession(channel, remoteAddress, parser.next());
+            if (deviceSession == null) {
+                return null;
+            }
+
+            position.setDeviceId(deviceSession.getDeviceId());
+
+            DateFormat dateFormat = new SimpleDateFormat("ddMMyyHHmmss");
+            dateFormat.setTimeZone(TimeZone.getTimeZone("UTC"));
+
+            String date = String.format("%06d", parser.nextInt());
+            String time = String.format("%06d", parser.nextInt());
+
+            position.setTime(dateFormat.parse(date + time));
+
+            position.setLatitude(convertCoordinates(parser.nextLong()));
+            position.setLongitude(convertCoordinates(parser.nextLong()));
+            position.setSpeed(UnitsConverter.knotsFromKph(parser.nextInt() / 100));
+            position.setCourse(parser.nextInt() / 100);
+            position.setAltitude(0);
+
+            position.set(Position.KEY_ODOMETER, parser.nextInt());
+
+            String input = new StringBuilder(String.format("%08d",
+                    Integer.parseInt(Integer.toString(parser.nextInt(), 2)))).reverse().toString();

```

Comment:
What does this do? If it's a bitmask please use proper bit manipulation. Traccar has a utility class for that.

Reply:
At where? Example please!

Reply:
Search for BitUtil usage in other decoders.

Reply:
Done this! Commit later!